src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -573,21 +573,40 @@ void MPS::apply_swap_internal(uint_t index_A, uint_t index_B, bool swap_gate) {
 void MPS::apply_2_qubit_gate(uint_t index_A, uint_t index_B, Gates gate_type, const cmatrix_t &mat, bool is_diagonal)
 {
   // We first move the two qubits to be in consecutive positions
-  // If index_B > index_A, we move the qubit at index_B to index_A+1
-  // If index_B < index_A, we move the qubit at index_B to index_A-1, and then
-  // swap between the qubits
+  // For performance reasons, we move the qubit with higher bond dimension
+  // toward the qubit with lower bond dimension
+  // Case i:
+  //     If index_B > index_A, we either move the qubit at index_B to index_A+1
+  //                                 or  move the qubit at index_A to index_B-1
+  // Case ii:
+  //     If index_B < index_A, we either move the qubit at index_B to index_A-1, 
+  //                                  or move the qubit at index_A to index_B+1
+  //     Finally, A = min(index_A, index_B)
+
   uint_t A = index_A;
 
   bool swapped = false;
-
   if (index_B > index_A+1) {
-    change_position(index_B, index_A+1);  // Move B to be right after A
+    if (lambda_reg_[index_A] <= lambda_reg_[index_B-1]) {
+      change_position(index_B, index_A+1);  // Move B to be right after A
+    } else {
+      change_position(index_A, index_B-1);  // Move A to be right before B
+      index_A = index_B-1;
+    }
+
   } else if (index_A > 0 && index_B < index_A-1) {
-    change_position(index_B, index_A-1);  // Move B to be right before A
+    if (lambda_reg_[index_B] <= lambda_reg_[index_A-1]) {
+      change_position(index_A, index_B+1);  // Move A to be right after B
+    }  else {
+      change_position(index_B, index_A-1);  //Move B to be right before A
+    }
   }
+  // if B < A, swap A and B
```

Comment:
This comment is confusing, I thought you were performing a swap operation between the qubits